service/view_building_coordinator.cc

```diff
@@ -0,0 +1,183 @@
+        }
+    }
+
+    if (!cmuts.empty()) {
+        auto cmd = _group0.client().prepare_command(write_mutations{
+            .mutations{std::move(cmuts)},
+        }, guard, "update view building coordinator state");
+        co_await _group0.client().add_entry(std::move(cmd), std::move(guard), _as);
+        co_return std::nullopt;
+    }
+    co_return state;
+}
+
+std::set<view_name> view_building_coordinator::get_views_to_add(const vbc_state& state, const std::vector<view_name>& views, const std::vector<view_name>& built) {
+    std::set<view_name> views_to_add;
+    for (auto& view: views) {
+        if (!_db.find_keyspace(view.first).uses_tablets() || std::find(built.begin(), built.end(), view) != built.end()) {
+            continue;
+        }
+
+        auto base_id = get_base_id(view);
+        if (!state.tasks.contains(base_id) || !state.tasks.at(base_id).contains(view)) {
+            views_to_add.insert(view);
+        }
+    }
+    return views_to_add;
+}
+
+std::set<view_name> view_building_coordinator::get_views_to_remove(const vbc_state& state, const std::vector<view_name>& views) {
+    std::set<view_name> views_to_remove;
+    for (auto& [_, view_tasks]: state.tasks) {
+        for (auto& [view, _]: view_tasks) {
+            if (std::find(views.begin(), views.end(), view) == views.end()) {
+                views_to_remove.insert(view);
+            }
+        }
+    }
+    return views_to_remove;
+}
+
+future<std::vector<canonical_mutation>> view_building_coordinator::add_view(const group0_guard& guard, const view_name& view_name) {
+    vbc_logger.info("Register new view: {}.{}", view_name.first, view_name.second);
+
+    auto base_id = get_base_id(view_name);
+    auto& base_cf = _db.find_column_family(base_id);
+    auto erm = base_cf.get_effective_replication_map();
+    auto& tablet_map = erm->get_token_metadata().tablets().get_tablet_map(base_id);
+
+    std::vector<canonical_mutation> muts;
+    for (auto tid = std::optional(tablet_map.first_tablet()); tid; tid = tablet_map.next_tablet(*tid)) {
+        const auto& tablet_info = tablet_map.get_tablet_info(*tid);
+        auto range = tablet_map.get_token_range(*tid);
+
+        for (auto& replica: tablet_info.replicas) {
+            auto mut = co_await _sys_ks.make_vbc_task_mutation(guard.write_timestamp(), view_name, replica.host, replica.shard, range);
+            muts.emplace_back(std::move((mut)));
+        }
+    }
+    co_return muts;
+}
+
+future<std::vector<canonical_mutation>> view_building_coordinator::remove_view(const group0_guard& guard, const view_name& view_name) {
+    vbc_logger.info("Unregister all remaining tasks for view: {}.{}", view_name.first, view_name.second);
+    
+    auto muts = co_await _sys_ks.make_vbc_remove_view_tasks_mutations(guard.write_timestamp(), view_name);
+    co_return std::vector<canonical_mutation>{muts.begin(), muts.end()};
```

Comment:
Here, you should also make sure to remove the relevant rows from `built_tablet_views`.